js/README.md

```diff
@@ -17,50 +17,165 @@
+   4,    4,  bbb
+   5,    5, cccc
+*/
+```
+
+## Create a Table when the Arrow file is split across buffers
+
+```es6
+import { readFileSync } from 'fs';
+import { Table } from 'apache-arrow';
+
+const table = Table.from(...[
+    'latlong/schema.arrow',
+    'latlong/records.arrow'
+].map((file) => readFileSync(file)));
+
+console.log(table.toString());
+
+/*
+        origin_lat,         origin_lon
+35.393089294433594,  -97.6007308959961
+35.393089294433594,  -97.6007308959961
+35.393089294433594,  -97.6007308959961
+29.533695220947266, -98.46977996826172
+29.533695220947266, -98.46977996826172
+*/
 ```
 
-### Usage
-The library is designed to be used with node.js or in the browser, this repository contains examples of both.
+## Columns are what you'd expect
 
-#### Node
-Import the arrow module:
+```es6
+import { readFileSync } from 'fs';
+import { Table } from 'apache-arrow';
 
-``` js
-var arrow = require("arrow");
+const table = Table.from(...[
+    'latlong/schema.arrow',
+    'latlong/records.arrow'
+].map(readFileSync));
+
+const column = table.getColumn('origin_lat');
+const typed = column.slice();
+
+assert(typed instanceof Float32Array);
+
+for (let i = -1, n = column.length; ++i < n;) {
+    assert(column.get(i) === typed[i]);
+}
 ```
 
-See [bin/arrow_schema.js](bin/arrow_schema.js) and [bin/arrow2csv.js](bin/arrow2csv.js) for usage examples.
+## Use with MapD Core
+
+```es6
+import MapD from 'rxjs-mapd';
+import { Table } from 'apache-arrow';
 
-#### Browser
-Include `_bundles/arrow.js` in a `<script />` tag:
-``` html
-<script src="_bundles/arrow.js"/>
+const port = 9091;
+const host = `localhost`;
+const encrypted = false;
+const username = `mapd`;
+const password = `HyperInteractive`;
+const dbName = `mapd`, timeout = 5000;
+
+MapD.open(host, port, encrypted)
+    .connect(dbName, username, password, timeout)
+    .flatMap((session) =>
+        session.queryDF(`
+            SELECT origin_city
+            FROM flights
+            WHERE dest_city ILIKE 'dallas'
+            LIMIT 5`
+        ).disconnect()
+  )
+  .map(([schema, records]) =>
+      Table.from(schema, records))
+  .subscribe((table) => console.log(
+      table.toString({ index: true })));
+/*
+Index,   origin_city
+    0, Oklahoma City
+    1, Oklahoma City
+    2, Oklahoma City
+    3,   San Antonio
+    4,   San Antonio
+*/
 ```
-See [examples/read_file.html](examples/read_file.html) for a usage example.
 
-### API
-##### `arrow.getReader(buffer)`
-Returns an `ArrowReader` object representing the Arrow file or stream contained in
-the `buffer`.
+# Contribute
+
+See [develop.md](https://github.com/graphistry/arrow/blob/master/develop.md)
+
+Please create an issue if you encounter any bugs!
+
+PR's welcome! Here's some ideas:
+
+* API docs
+* More Tests/Benchmarks
+* Performance optimizations
+* Arrows from node-streams and async-iterators
+* GPU Arrows from node-opencl and node-cuda buffers
+* Bindings to [libgdf](https://github.com/gpuopenanalytics/libgdf)
```

Comment:
We should separately start a discussion with GoAI about merging efforts with libgdf and the arrow_gpu package. Maintaining IPC code and analytics kernels in multiple places may result in a fragmentation of efforts